collector/writer.go

```diff
@@ -29,56 +32,108 @@ func (w CollectdWriter) Write(s Stats) error {
-		"cpu.total":  s.Stats.CPUStats.CPUUsage.TotalUsage,
-
-		"memory.limit": s.Stats.MemoryStats.Limit,
-		"memory.max":   s.Stats.MemoryStats.MaxUsage,
-		"memory.usage": s.Stats.MemoryStats.Usage,
-
-		"memory.active_anon":   s.Stats.MemoryStats.Stats.TotalActiveAnon,
-		"memory.active_file":   s.Stats.MemoryStats.Stats.TotalActiveFile,
-		"memory.cache":         s.Stats.MemoryStats.Stats.TotalCache,
-		"memory.inactive_anon": s.Stats.MemoryStats.Stats.TotalInactiveAnon,
-		"memory.inactive_file": s.Stats.MemoryStats.Stats.TotalInactiveFile,
-		"memory.mapped_file":   s.Stats.MemoryStats.Stats.TotalMappedFile,
-		"memory.pg_fault":      s.Stats.MemoryStats.Stats.TotalPgfault,
-		"memory.pg_in":         s.Stats.MemoryStats.Stats.TotalPgpgin,
-		"memory.pg_out":        s.Stats.MemoryStats.Stats.TotalPgpgout,
-		"memory.rss":           s.Stats.MemoryStats.Stats.TotalRss,
-		"memory.rss_huge":      s.Stats.MemoryStats.Stats.TotalRssHuge,
-		"memory.unevictable":   s.Stats.MemoryStats.Stats.TotalUnevictable,
-		"memory.writeback":     s.Stats.MemoryStats.Stats.TotalWriteback,
+	totalPercent, userPercent, systemPercent := calculateCPUPercent(s.Stats.PreCPUStats, s.Stats.CPUStats)
+	cpu_metrics := map[string]float64{
+		"vcpu-user":   userPercent,
+		"vcpu-system": systemPercent,
+		"vcpu-total":  totalPercent,
 	}
 
+	memory_metrics := map[string]uint64{
+		"memory-limit": s.Stats.MemoryStats.Limit,
+		"memory-max":   s.Stats.MemoryStats.MaxUsage,
+		"memory-usage": s.Stats.MemoryStats.Usage,
+
+		"memory-active_anon":   s.Stats.MemoryStats.Stats.TotalActiveAnon,
+		"memory-active_file":   s.Stats.MemoryStats.Stats.TotalActiveFile,
+		"memory-cache":         s.Stats.MemoryStats.Stats.TotalCache,
+		"memory-inactive_anon": s.Stats.MemoryStats.Stats.TotalInactiveAnon,
+		"memory-inactive_file": s.Stats.MemoryStats.Stats.TotalInactiveFile,
+		"memory-mapped_file":   s.Stats.MemoryStats.Stats.TotalMappedFile,
+		"memory-pg_fault":      s.Stats.MemoryStats.Stats.TotalPgfault,
+		"memory-pg_in":         s.Stats.MemoryStats.Stats.TotalPgpgin,
+		"memory-pg_out":        s.Stats.MemoryStats.Stats.TotalPgpgout,
+		"memory-rss":           s.Stats.MemoryStats.Stats.TotalRss,
+		"memory-rss_huge":      s.Stats.MemoryStats.Stats.TotalRssHuge,
+		"memory-unevictable":   s.Stats.MemoryStats.Stats.TotalUnevictable,
+		"memory-writeback":     s.Stats.MemoryStats.Stats.TotalWriteback,
+	}
+
+	network_metrics := map[string]string{}
+
 	for _, network := range s.Stats.Networks {
-		metrics["net.rx_bytes"] += network.RxBytes
-		metrics["net.rx_dropped"] += network.RxDropped
-		metrics["net.rx_errors"] += network.RxErrors
-		metrics["net.rx_packets"] += network.RxPackets
-
-		metrics["net.tx_bytes"] += network.TxBytes
-		metrics["net.tx_dropped"] += network.TxDropped
-		metrics["net.tx_errors"] += network.TxErrors
-		metrics["net.tx_packets"] += network.TxPackets
+		network_metrics["if_dropped"] = fmt.Sprintf("%d:%d", network.RxDropped, network.TxDropped)
```

Comment:
Yeah if you look at the types db here is an example:

```if_dropped              rx:DERIVE:0:U, tx:DERIVE:0:U```

When you send the metric it then breaks down the type into various subtypes, which gives you nesting when looking at the hierarchy.   So it looks like... if_dropped.rx instead of if_dropped-rx.  It has the added benefit of being less individual metrics to push.